dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -1270,4 +1421,72 @@ private boolean userHasAccess( SqlRowSet rowSet )
             throw new IllegalArgumentException( e );
         }
     }
+
+    private void populateCaches( EventSearchParams params, String sql, boolean isSuperUser )
```

Comment:
It seems like we run the query one more time to populate the cache? That sounds a bit excessive.

Reply:
This is a tradeoff. The query is run twice, yes. But only if the idScheme is Code or Attribute. On the other side, we do not run multiple queries when fetching each item from the cache because we have prepopulated cache. So it is a tradeoff. I think 1 query (even heavier) is better than 5, 10, 20 small queries.